packages/components/src/inputs/src/TextArea.tsx

```diff
@@ -0,0 +1,294 @@
+        } else if (maxRows && currentRowsWithText >= maxRows) {
+            // if the number of rows with text is greater than or equal to the max rows, we need to adjust the height
+            setRows(maxRows);
+        } else {
+            setRows(Math.max(currentRowsWithText, DefaultMinimumTextAreaRows));
+        }
+    }, [mergedTextAreaRef, rowHeight, maxRows, rowsProp]);
+
+    // adjustRows needs to be called here instead of in handleTextChanged because handleTextChanged is not called when there is a defaultValue on load.
+    // truncateText also needs to be here so that if the default text goes over the maxLength, it is truncated.
+    useIsomorphicLayoutEffect(() => {
+        const newValue = truncateText(value, maxLength);
+        onChange(newValue);
+        adjustRows();
+    }, [value, adjustRows]);
+
+    const inputMarkup = (
+        <ClearContainerSlots>
+            <InputGroup
+                isFluid
+                size={size}
+                className={styles["hop-TextArea__InputGroup"]}
+                isDisabled={isDisabled}
+                isInvalid={isInvalid}
+                inputClassName={styles["hop-TextArea__textarea"]}
+            >
+                <RACTextArea ref={mergedTextAreaRef} placeholder={placeholder} cols={cols} rows={rows} />
+
+                {showCharacterCount && maxLength && <CharacterCount charactersLeft={maxLength - characterCount} />}
+            </InputGroup>
+        </ClearContainerSlots>
+    );
+
+    const childrenMarkup = composeRenderProps(children, prev => {
+        return (
+            <>
+                <SlotProvider values={[
+                    [LabelContext, { className: styles["hop-TextArea__Label"] }],
+                    [HelperMessageContext, { className: styles["hop-TextArea__HelperMessage"] }],
+                    [ErrorMessageContext, { className: styles["hop-TextArea__ErrorMessage"] }]
+                ]}
+                >
+                    {prev}
+                </SlotProvider>
+                {inputMarkup}
+            </>
+        );
+    });
+
+    return (
+        <RACTextField
+            ref={ref}
+            value={value}
+            style={style}
+            className={classNames}
+            maxLength={maxLength}
+            onChange={onChange}
+            isDisabled={isDisabled}
+            isInvalid={isInvalid}
+            data-resize-mode={resizeMode}
+            {...otherProps}
+        >
+            {childrenMarkup}
+        </RACTextField>
+    );
+}
+
+interface CharacterCountProps {
+    charactersLeft: number;
+}
+
+function CharacterCount({ charactersLeft }: CharacterCountProps) {
+    const stringFormatter = useLocalizedString();
+
+    const accessibilityString = stringFormatter.format("Input.charactersLeft", { charLeft: charactersLeft });
+
+    return <Text aria-label={accessibilityString} color="neutral-weakest" size="xs" className={styles["hop-TextArea__char-count"]}>{charactersLeft}</Text>;
```

Comment:
Is there a reason for the color being a style prop and the charCount classnames?

Reply:
hmm didn't realize. The color prop was just there before, but I needed to add some extra styling so I added the class.
I can remove the style prop and put it in the css.